### PR TITLE
Drop the adopt state seed and key mdns claims on the device-named PTR

### LINKED ISSUE #2389: Renamed adopt seeds an un-withdrawable mdns ONLINE claim until the flash
## Summary

Adopting an importable device under a different name than its factory broadcast seeds an mdns ONLINE claim that nothing can withdraw until the device is flashed. If the user never flashes (or the device disappears before flashing), the adopted entry sits ONLINE indefinitely.

## Mechanism

The adopt seed (`controllers/devices/importable.py`) applies `ONLINE` under the **chosen** YAML name with `claim=True` while the device still broadcasts under its factory `mdns_name`. In the rename-during-adopt case there is no PTR for the chosen name, so no browser `Removed` can fire for it, `should_ping` skips the mdns-owned entry, and the claim is un-withdrawable until the post-flash boot announce under the new name re-enters the normal browser lifecycle.

The no-rename case is fine: the chosen name matches the factory broadcast, a live PTR exists, and the `Removed` lifecycle covers it.

Pre-existing before #2386 (the sweep used to ping the entry and drop the OFFLINE verdict at the priority gate). Bounded by the adopt flow in intent, unbounded in time.

## Direction to evaluate

Seed the renamed-adopt case under the `ping` source instead of `mdns` (the ICMP sweep then owns liveness until the flashed firmware announces), or track the seed and withdraw it via `source_withdrawn` when the factory-name service goes away.

### PR DESCRIPTION
**Pull request by bdraco**
# What does this implement/fix?

Close the last PTR-less mdns claim, following #2393 (now rebased onto its merge). Adopting an importable device under a different name than its factory broadcast seeded an mdns ONLINE claim under the chosen name; no PTR exists for that name, so no `Removed` could ever withdraw it, and a renamed adopt that never flashed sat ONLINE indefinitely.

Two changes:

- The adopt path seeds no state at all — the real sources decide. Discovery is mDNS-based, so the device was broadcasting milliseconds ago: a same-name adopt claims ONLINE via the esphomelib probe's synchronous cache hit in the same call, and a rename-during-adopt (whose chosen name has no PTR) gets the factory broadcast's cached IP applied and is arbitrated by the regular ping sweep instead of holding a fabricated ONLINE. The drawer never shows a ping that didn't run.
- `_apply_service_info`'s claim gate now keys on the **device-named** service's PTR — the one whose `Removed` maps back to the device — instead of the resolved service's. The post-adopt probe rides the factory service (whose PTR is live), and under the old gate that cross-name resolve would have re-created the un-withdrawable claim; it now applies its data (version / config_hash / IP, the probe's purpose) and takes no ownership. For every other caller the resolved service and the device name agree, so nothing else changes.

With this, no PTR-less mdns claim path remains, and the CLAUDE.md invariant bullet drops its last carve-out.

Also carries the accepted review follow-ups from #2393's final round (maintainer's call moved them here rather than blocking that merge): the drawer's TXT panel picks live esphomelib TXT first, then live `_http._tcp` identity TXT, then the aged records (so a device recompiled without `api:` doesn't show a dead version forever, and an all-expired cache keeps last-known identity instead of emptying the panel); debug logs when a successful resolve is denied the mdns claim for lack of a live PTR (both gated claim sites); and the three http-withdrawal tests now stub the PTR lookup explicitly instead of riding `_zeroconf is None`. The proxy-served-PTR deferral edge case was reviewed and deliberately deferred (#2406).

**Related issue or feature (if applicable):**

- fixes https://github.com/esphome/device-builder/issues/2389

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue) — `bugfix`
- [ ] New feature (non-breaking change which adds functionality) — `new-feature`
- [ ] Enhancement to an existing feature — `enhancement`
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected) — `breaking-change`
- [ ] Refactor (no behaviour change) — `refactor`
- [ ] Documentation only — `docs`
- [ ] Maintenance / chore — `maintenance`
- [ ] CI / workflow change — `ci`
- [ ] Dependencies bump — `dependencies`

## Frontend coordination

- [x] No frontend change needed
- [ ] Companion frontend PR: esphome/device-builder-frontend#<number>

## Checklist

- [x] The code change is tested and works locally.
- [x] Pre-commit hooks pass (`ruff`, `codespell`, yaml/json/python checks).
- [x] Tests have been added or updated under `tests/` where applicable.
- [x] `components.index.json` / `definitions/components/*.json` have **not** been hand-edited (regenerate via `script/sync_components.py` if a sync is needed).
- [x] Architecture-level changes are reflected in `docs/ARCHITECTURE.md` and/or `docs/API.md`.

### CODE REVIEW

**Review by esphbot — APPROVED** [review]
> [!TIP]
> No blocking issues found — ready to merge.

**Review by esphbot — APPROVED** [review]
> [!TIP]
> No blocking issues found — ready to merge.

**Review by esphbot — APPROVED** [review]
> [!TIP]
> No blocking issues found — ready to merge.

**Review by esphbot — APPROVED** [review]
> [!TIP]
> No blocking issues found — ready to merge.

**Review by esphbot — APPROVED** [review]
> [!TIP]
> No blocking issues found — ready to merge.

File: esphome_device_builder/controllers/devices/importable.py [hunk 1]
```diff
@@ -151,9 +151,13 @@ def _cleanup() -> None:
     # Skip-the-wait state seed; the device was advertising on
     # mDNS milliseconds ago, so pin ONLINE + the cached IP now
     # rather than blinking through OFFLINE for ~10s waiting on
-    # the next ping sweep. Probe esphomelib too so version /
+    # the next ping sweep. Seeded under ``ping`` — for a
+    # rename-during-adopt no PTR exists under *name*, so an mdns
+    # claim would have no ``Removed`` to withdraw it (#2389); the
+    # sweep keeps arbitrating until the flashed firmware announces
+    # and mdns takes over. Probe esphomelib too so version /
```
[the remark this answers is not in the document]
Comment (bdraco) [reply esphome_device_builder/controllers/devices/importable.py]: Fixed in f24d6e035 — the issue reference is dropped from the comment. The lowercase `mdns` in that sentence is deliberate: it names the reachability source (`mdns` in the precedence ledger), while `mDNS` in the surrounding lines names the protocol; both usages follow the monitor package's convention.

**Review by esphbot — APPROVED** [review]
> [!TIP]
> No blocking issues found — ready to merge.

File: esphome_device_builder/controllers/devices/importable.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (esphbot) [reply esphome_device_builder/controllers/devices/importable.py]: Agreed on `mdns` vs `mDNS` — lowercase names the ledger source, matching `shared.py`'s `_SOURCE_PRIORITY` keys and the new debug string. On the dropped `#2389`: the repo bars "closes #N" cross-refs but sanctions an issue number as an empirical anchor, and neighbouring mDNS code is anchored that way (#2369, #1776, #1748, #2388). Removing it costs the only pointer to why the seed can't be mdns-sourced. Author's call; not blocking.

**Review by esphbot — APPROVED** [review]
> [!TIP]
> No blocking issues found — ready to merge.

**Review by esphbot — APPROVED** [review]
> [!TIP]
> No blocking issues found — ready to merge.

**Review by esphbot — APPROVED** [review]
> [!TIP]
> No blocking issues found — ready to merge.

**Review by esphbot — APPROVED** [review]
> [!TIP]
> No blocking issues found — ready to merge.
